test-data/unit/stubgen.test

```diff
@@ -698,6 +698,53 @@ class Y(NamedTuple):
     a: int
     b: str
 
+[case testNamedTupleClassSyntax_semanal]
+from typing import NamedTuple
+
+class A(NamedTuple):
+    x: int
+    y: str = 'a'
+
+class B(A):
+    z1: str
+    z2 = 1
+    z3: str = 'b'
+
+class RegularClass:
+    x: int
+    y: str = 'a'
+    class NestedNamedTuple(NamedTuple):
+        x: int
+        y: str = 'a'
+    z: str = 'b'
+
+# TODO: make sure that nested classes in `NamedTuple` are supported:
+class NamedTupleWithNestedClass(NamedTuple):
+    class Nested:
+        x: int
+        y: str = 'a'
```

Comment:
Should this be put into a separate xfail test? Currently it feels like we're asserting incorrect behaviour here

Reply:
I will work on it right after this PR, so it does not really matter.

Reply:
I don't really like asserting incorrect behaviour in a test even temporarily (I've often planned to work on something immediately after, but then got sidetracked/forgot about it/found it was harder than I anticipated :)